server/service/integration_mdm_test.go

```diff
@@ -520,6 +521,16 @@ func (s *integrationMDMTestSuite) SetupSuite() {
 		_, _ = w.Write([]byte(fmt.Sprintf(`{"results": [%s]}`, strings.Join(objs, ","))))
 	}))
 
+	s.appleGDMFSrv = httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		// load the test data from the file
+		b, err := os.ReadFile("../mdm/apple/gdmf/test_data.json")
+		require.NoError(s.T(), err)
+		_, err = w.Write(b)
+		require.NoError(s.T(), err)
+	}))
+
+	s.T().Setenv("FLEET_DEV_GDMF_URL", s.appleGDMFSrv.URL)
```

Comment:
Do we already have enrollment integration tests that will automatically test that behavior?

Reply:
I have a todo to follow up with more integration test coverage. It's a bit more work because it requires updating the mock device client to include Apple's custom `x-apple-aspen-deviceinfo` header so I was going to try to work on that during the freeze period.  As it stands, the enrollment integration tests will implicitly test the case where there is no match in the gdmf assets (current business logic is to allow the device to proceed with enrollment in that case).